lib/AST/FrontendSourceFileDepGraphFactory.cpp

```diff
@@ -406,26 +406,19 @@ class UsedDeclEnumerator {
   StringRef swiftDeps;
 
   /// Cache these for efficiency
-  const DependencyKey sourceFileInterface;
   const DependencyKey sourceFileImplementation;
 
-  function_ref<void(const DependencyKey &, const DependencyKey &)> createDefUse;
-
 public:
-  UsedDeclEnumerator(
-      const SourceFile *SF, const DependencyTracker &depTracker,
-      StringRef swiftDeps,
-      function_ref<void(const DependencyKey &, const DependencyKey &)>
-          createDefUse)
+  UsedDeclEnumerator(const SourceFile *SF, const DependencyTracker &depTracker,
+                     StringRef swiftDeps)
```

Comment:
What's the advantage in passing the function ref down recursively vs storing it in the instance?
Since you did it, I bet there is a good reason. Would be nice to have a comment, though.

Reply:
`llvm::function_ref` is non-owning. It makes me nervous to see it stored.

Reply:
D'Oh! Thank you for fixing this!